### PR TITLE
Translation fengqifang

### PR DESCRIPTION
**Pull request by fengqifang**
Lines 1-85 for data.table translation to Simplified Chinese

### CODE REVIEW

**Comment by MichaelChirico** [comment]
hi @fengqifang thanks so much!!

the target for this pull request is the master branch, could you edit it so that it's targeting the translation branch instead?

**Comment by MichaelChirico** [comment]
Thanks Fengqi! @Shayebuhuii can you please review?

Fengqi feel free to add the C portion (lines 1-163 of zh_CN.po) here or in a separate PR, and of course take your time 😄

**Comment by MichaelChirico** [comment]
@sunshine1126 do you have any comments for `po/R-zh_CN.po`? I notice many comments for `zh_CN.po` but none for `R-zh_CN.po`

**Comment by sunshine1126** [comment]
@MichaelChirico Sorry, I omitted the comments for po/R-zh_CN.po and I will review it today. Thanks for  your prompting.

File: po/R-zh_CN.po [hunk 1]
```diff
@@ -11,65 +11,65 @@ msgstr ""
...
 
 msgid "Argument 'value.name' should not overlap with column names in result:"
-msgstr ""
+msgstr "参数'value.name'不能与结果中已有列名重复"
 
 msgid ""
 "POSIXlt column type detected and converted to POSIXct. We do not recommend "
 "use of POSIXlt at all because it uses 40 bytes to store one date."
-msgstr ""
+msgstr "检测到使用了或转成POSIXlt类型的列，data.table不建议使用POSIXlt对象, 因为其使用40bytes存储一个日期"
```
Comment (sunshine1126) [inline po/R-zh_CN.po]: “检测到POSIXlt列的类型，并将其转换为POSIXct。我们完全不建议使用POSIXlt，因为它使用40个字节来存储一个日期。”

File: po/R-zh_CN.po [hunk 2]
```diff
@@ -11,65 +11,65 @@ msgstr ""
...
 
 msgid "Item"
-msgstr ""
+msgstr "条目"
 
 msgid "has"
 msgstr ""
```
Comment (sunshine1126) [inline po/R-zh_CN.po]: “具有”

File: po/R-zh_CN.po [hunk 3]
```diff
@@ -78,13 +78,13 @@ msgid "rows but longest item has"
...
```
Comment (sunshine1126) [inline po/R-zh_CN.po]: “行，但最长条目有”

File: po/R-zh_CN.po [hunk 4]
```diff
@@ -78,13 +78,13 @@ msgid "rows but longest item has"
...
 
 msgid "; recycled with remainder."
-msgstr ""
+msgstr "用余数循环填充"
 
 msgid "has 0 rows but longest item has"
 msgstr ""
```
Comment (sunshine1126) [inline po/R-zh_CN.po]: ”没有行，但最长的条目有“

File: po/R-zh_CN.po [hunk 5]
```diff
@@ -17,7 +17,7 @@ msgid ""
 "Internal error -- difftime objects may not be added to IDate, but Ops "
 "dispatch should have intervened to prevent this"
 msgstr "内部错误 -- 'difftime'对象不能与IDate类型相加"
```
Comment (sunshine1126) [inline po/R-zh_CN.po]: "内部错误 -- 'difftime'对象可能不能被添加到IDate类型"

File: po/R-zh_CN.po [hunk 6]
```diff
@@ -11,65 +11,65 @@ msgstr ""
...
 
 msgid "data.table package loaded. When developing don't load package"
-msgstr ""
+msgstr "data.table包已载入, 开发时无需载入"
 
 msgid ""
 "Internal error -- difftime objects may not be added to IDate, but Ops "
 "dispatch should have intervened to prevent this"
-msgstr ""
+msgstr "内部错误 -- 'difftime'对象不能与IDate类型相加, 正确操作应该防止此行为发生"
```
Comment (sunshine1126) [inline po/R-zh_CN.po]: "内部错误 -- 'difftime'对象可能不能被添加到IDate类型，但正确操作应该防止此行为发生"

**Comment by MichaelChirico** [comment]
Thanks for the thorough review!

> There are three versions of the translation

I am not sure what you mean by three versions, could you help clarify?

**Comment by sunshine1126** [comment]
> I am not sure what you mean by three versions, could you help clarify?
@fengqifang may modify the previous version, but not all translations can be commented on in the latest version. @MichaelChirico

**Comment by sunshine1126** [comment]
> I am not sure what you mean by three versions, could you help clarify?

@fengqifang may modify the previous version, but not all translations can be commented on in the latest version. @MichaelChirico

**Comment by MichaelChirico** [comment]
Still not sure I follow 100%, but perhaps has to do with the `Outdated` comments -- this happens because Fengqi has pushed a new commit:

https://github.com/Rdatatable/data.table/pull/3990/commits/b3bce0c952c3d782ef25209a6b65eac67fa7daee

After that the review comments that were made on lines that have changed become outdated (this is because your comments have been addressed & incorporated).

The current/latest version of the translation can always be viewed on the `Files Changed` tab:

https://github.com/Rdatatable/data.table/pull/3990/files

Hope that helps

**Comment by sunshine1126** [comment]
> The current/latest version of the translation can always be viewed on the `Files Changed` tab

I see, thanks for your answer. @MichaelChirico

**Comment by MichaelChirico** [comment]
Hey @fengqifang please have a look at the outstanding items raised by Ping Yu when you get a chance. Thanks!

**Comment by MichaelChirico** [comment]
Hey @fengqifang sorry, something appears to have gone wrong. The diff on your files is huge now after your latest commit, could you double check?

It's possible it's an encoding issue... if so, you could try copy-pasting your updates into the UI here.

**Comment by fengqifang** [comment]
> Hey @fengqifang sorry, something appears to have gone wrong. The diff on your files is huge now after your latest commit, could you double check?
> 
> It's possible it's an encoding issue... if so, you could try copy-pasting your updates into the UI here.

I noticed it also,  

> Hey @fengqifang sorry, something appears to have gone wrong. The diff on your files is huge now after your latest commit, could you double check?
> 
> It's possible it's an encoding issue... if so, you could try copy-pasting your updates into the UI here.

En, I noticed something wrong.

I just updated the R-zh_CN.po, it seems that my local git version is messed up. 

I attached the update .po file (txt) here, hope that helps.

msgid ""
msgstr ""
"Project-Id-Version: data.table 1.12.5\n"
"POT-Creation-Date: 2019-10-21 16:15\n"
"PO-Revision-Date: 2019-10-04 17:06+08\n"
"Last-Translator: Michael Chirico <MichaelChirico4@gmail.com>\n"
"Language-Team: Mandarin\n"
"Language: Mandarin\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"

msgid "data.table package loaded. When developing don't load package"
msgstr "data.table包已载入, 开发时无需载入"

msgid ""
"Internal error -- difftime objects may not be added to IDate, but Ops "
"dispatch should have intervened to prevent this"
msgstr "内部错误 -- 'difftime'对象可能不能被添加到IDate类型"
"但是正确操作应该防止此行为发生"

msgid "binary + is not defined for \"IDate\" objects"
msgstr "没有为\"IDate\" 对象定义的'+'操作"

msgid "can only subtract from \"IDate\" objects"
msgstr "只能从\"IDate\"对象中减去"

msgid "Internal error: storage mode of IDate is somehow no longer integer"
msgstr "内部错误： IDate的存储模式不为整型"

msgid "unary - is not defined for \"IDate\" objects"
msgstr "没有为\"IDate\" 对象定义的一元的'-'操作"

msgid ""
"Internal error -- difftime objects may not be subtracted from IDate, but Ops "
"dispatch should have intervened to prevent this"
msgstr "内部错误 -- 不能与IDate类型中减去'difftime'对象"
"但是正确操作应该防止此行为发生"

msgid "Valid options for ms are 'truncate',"
msgstr "'ms'有效的选项为'truncate'"

msgid "'nearest', and 'ceil'."
msgstr "'nearest'和'ceil'"

msgid ""
"as.data.table.array method should only be called for arrays with 3+ "
"dimensions; use the matrix method for 2-dimensional arrays"
msgstr "请只对3维或以上的数组使用as.data.table.array方法"
"对二维数组请使用matrix方法"

msgid ""
"Argument 'value.name' must be scalar character, non-NA and at least one "
"character"
msgstr "参数'value.name'必须为标量字符或non-NA"
"至少包含一个字符"

msgid "Argument 'sorted' must be scalar logical and non-NA"
msgstr "参数'sorted'必须为标量逻辑或non-NA"

msgid "Argument 'na.rm' must be scalar logical and non-NA"
msgstr "参数'na.rm'必须为标量逻辑或non-NA"

msgid "Please provide either 'key' or 'sorted', but not both."
msgstr "请提供参数'key'或'sorted',但不可同时提供"

msgid "Argument 'value.name' should not overlap with column names in result:"
msgstr "参数'value.name'不能与结果中已有列名重复"

msgid ""
"POSIXlt column type detected and converted to POSIXct. We do not recommend "
"use of POSIXlt at all because it uses 40 bytes to store one date."
msgstr "检测到使用了POSIXlt类型的列， 并转成POSIXct"
"data.table不建议使用POSIXlt对象, 因为其使用40字节来存储一个日期"

msgid "Item"
msgstr "条目"

msgid "has"
msgstr "具有"

msgid "rows but longest item has"
msgstr "行，但最长条目有"

msgid "; recycled with remainder."
msgstr "用余数循环填充"

msgid "has 0 rows but longest item has"
msgstr "没有行，但最长的条目有"

msgid "; filled with NA"
msgstr "用NA填充"

**Comment by MichaelChirico** [comment]
Thanks @fengqifang I think that did it. I already had the previous commit loaded on `translation-fengqifang` locally, so I was able to copy-paste what you put above then add it and `git push --force` here to resolve.

I'm going to go ahead & merge. Thanks again @fengqifang and @sunshine1126 !!

**Pull request merged**